### PR TITLE
[AMDGPU] Add some more GFX12 test coverage

### PR DESCRIPTION
**Pull request by jayfoad**

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/mixed-vmem-types.ll [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  %21 = call i32 @llvm.amdgcn.raw.buffer.load.i32(<4 x i32> %12, i32 0, i32 0, i32 0)
+  %.not1 = icmp eq i32 %21, 2752
+  %22 = getelementptr i8, ptr addrspace(4) %3, i64 16
+  %23 = load <8 x i32>, ptr addrspace(4) %22, align 32
+  %24 = load <4 x i32>, ptr addrspace(4) %3, align 16
+  %25 = call float @llvm.amdgcn.image.sample.lz.2d.f32.f16(i32 1, half 0xHBC00, half 0xHBC00, <8 x i32> %23, <4 x i32> %24, i1 false, i32 0, i32 0)
+  %26 = fcmp oeq float %25, 1.000000e+00
+  %27 = call i32 @llvm.amdgcn.raw.buffer.load.i32(<4 x i32> %10, i32 0, i32 0, i32 0)
+  %.not2 = icmp eq i32 %27, 2752
+  %28 = select i1 %.not2, i1 %26, i1 false
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/mixed-vmem-types.ll:153]: Use named values

File: llvm/test/CodeGen/AMDGPU/fmaximum3.ll [hunk 2]
```diff
@@ -3766,3 +3766,69 @@ define <2 x double> @v_no_fmaximum3_f64__multi_use(double %a, double %b, double
   %insert.1 = insertelement <2 x double> %insert.0, double %max1, i32 1
   ret <2 x double> %insert.1
 }
+
+; Checks whether the test passes; performMinMaxCombine() should not optimize vector patterns of maximum3
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/fmaximum3.ll:3770]: This is achieved by v_fmaximum3_v2f16 already? This one just has a second instance

File: llvm/test/CodeGen/AMDGPU/mixed-vmem-types.ll (hunk 1 above)
Comment (jayfoad) [reply llvm/test/CodeGen/AMDGPU/mixed-vmem-types.ll]: Done

File: llvm/test/CodeGen/AMDGPU/fmaximum3.ll (hunk 2 above)
Comment (jayfoad) [reply llvm/test/CodeGen/AMDGPU/fmaximum3.ll]: Good point. This had been removed upstream but for some reason lived on in our downstream branch.

**Review by arsenm — APPROVED** [review]

**Pull request merged**